### ISSUE TITLE
Time to cut a release?

### ISSUE THREAD

**Issue by eileenmcnaughton** [civicrm/org.civicrm.volunteer#508]
@ginkgomzd - just saw a person hit a bug you fixed a long time ago https://civicrm.stackexchange.com/questions/30389/civivolunteer-error-after-upgrade-to-civi-5-11-0-on-joomla

but you haven't done a release for a long time either - might be time?

**Comment by kelizoliva** [comment]
Would love to see a release.

**Comment by ginkgomzd** [comment]
Yes, it is well passed time for a release.

After several set-backs today we kicked off a round of development for some customizations to a fork of CiviVolunteer. Part of the intent though is to resolve some known issues upstream (i.e. here), especially with the opportunity search.

What more immediately stands in the way of stamping the current master as a release are two things off the top of my head.:
 1) I'm pretty sure we have not ported up-stream a fix for an issue with forms caused by CiviCRM upgrading Angular in core.
2) Sufficient testing of master.

If anyone can attest to the viability of master as a release, and share their CiviCRM version and CMS version, it would be helpful.

Thanks

**Comment by kelizoliva** [comment]
I would be happy to test the master release on a few test sites this week, if that would be helpful. I should be able to test and provide updates on viability (here, and in respective issues), by Friday.

Curious about this fork; could you link to a thread with details on that, if one exists?

**Comment by ginkgomzd** [comment]
Thanks @kelizoliva, that will be a big help.

I just confirmed the fix I had in mind is already in master. Apparently, this bug has been resolved multiple times...

There isn't anything to share on the fork. The fork is just a place for private customizations that are not suitable for upstream. There is no intention to diverge any more than necessary, and we seek to release anything that can be released.

**Comment by jusfreeman** [comment]
Any chances of doing a release of org.civicrm.volunteer sometime soon? The last release was 2018-06-29 and I don't expect the user base is having much luck with CiviVolunteer on newer versions of CiviCRM and PHP 7.x.

We're patching for #477 (from August 2018) for PHP 7.x because of the errors generated.

**Comment by eileenmcnaughton** [comment]
@jusfreeman did you read above "If anyone can attest to the viability of master as a release, and share their CiviCRM version and CMS version, it would be helpful."

Maybe confirming your experience of using the master version on latest Civi will help

**Comment by kelizoliva** [comment]
On that note - I've been under a pile of work and haven't had a chance to sit down and test master as I said I would a couple of weeks ago. I am hopeful that I will finally get to it next week s a I have a few clients that rely on it and would like to see bugs resolved. Once I have been able to test for issues hopefully that will be enough to push this live, but I believe a second or third pair of eyes on master would be optimal.

**Comment by ginkgomzd** [comment]
I would still appreciate anyone validating master.
I will be doing my own testing soon... but I'm also wanting to put in a bug fix that doesn't exist yet. It's more a performance optimization than a bug fix, and I've only heard complaints from my own customers who arguably are not using the system as originally intended... so if someone nominates master as a release candidate, I won't delay the release

**Comment by ginkgomzd** [comment]
oh right, and I would also like to take care of 
`Deprecated function CRM_Core_OptionGroup::getValue, use CRM_Core_PseudoConstant::getKey.` before releasing.

**Comment by kelizoliva** [comment]
Tested the master branch on WordPress 5.2.2 with CiviCRM 5.13.4. Was able to perform all usual functions (create volunteer project, create opportunities, test volunteer sign-up, search volunteer opportunities) without any issues. 

Reviewed the logs and am not seeing any errors, particularly the permissions issue that we were previously seeing:

`Declaration of CRM_Volunteer_Permission::check($permissions) should be compatible with CRM_Core_Permission::check($permissions, $contactId = NULL) Permission.php:151`

From my perspective everything appears clear to release the next version!

**Comment by kelizoliva** [comment]
Tested this on Drupal 7.67 with CiviCRM 5.13.4. Same results as my WordPress testing, however the issue that I bring up in #494 still persists. The custom extension written to override the behavior continues to function as expected, so this will do for now. If others report a similar issue, may want to give that ticket some attention in a future release, though.

**Comment by karenannx** [comment]
What about the report, is that working? If so, very excited.

 

Best,

Karen-Ann

**Comment by kelizoliva** [comment]
@karenannx The volunteer report is working for me. I'm not sure what specific issue you are referring to - could you reference a ticket?

**Comment by kelizoliva** [comment]
Ah, its been a while. I see, you are referring to [this](https://civicrm.stackexchange.com/questions/30389/civivolunteer-error-after-upgrade-to-civi-5-11-0-on-joomla). I am unable to test with Joomla (and this is not something I have experienced with WordPress or Drupal), but if you are able to replace the civi volunteer directory with the master branch, perhaps you could test and report back?

**Comment by ginkgomzd** [comment]
> Tested this on Drupal 7.67 with CiviCRM 5.13.4. Same results as my WordPress testing, however the issue that I bring up in #494 still persists. The custom extension written to override the behavior continues to function as expected, so this will do for now. If others report a similar issue, may want to give that ticket some attention in a future release, though.

Hi @kelizoliva , yeah a resolution to #494 is actually #509 which I noted is non-trivial. We're definitely going to need to find a funder for that. If we get a new custom project, we might be able to squeeze it in, but at the moment I don't have one on the horizon.

**Comment by ginkgomzd** [comment]
Hi @karenannx ,
I do believe that the error about CRM_Core_Permission::check() is resolved in the master branch (and so in the upcoming release as well.
Cheers

**Comment by ginkgomzd** [comment]
Okay, 
Thanks for the testing @kelizoliva !
I've done some myself and I'm feeling pretty good about a release.

It's late in the day, and I am undecided about something to include in the release, so I'm going to sleep on it.

I just deployed a performance fix to the opportunity search. Wow, my head is spinning. I want to include it in the release, but it's a significant code change that hasn't been exercised much yet.

If you are inclined, you could test out the 2.5-rc branch and try to break the opportunity search.

Thanks again

**Comment by karenannx** [comment]
Thank you so much for letting me know Michael, any idea on lead time of release?

 

Best,

Karen-Ann

**Comment by ginkgomzd** [comment]
As soon as tomorrow, but you never know, so hopefully not later than this week.

**Comment by karenannx** [comment]
Thanks Michael, fingers crossed!

 

Best,

Karen-Ann

**Comment by kelizoliva** [comment]
> It's late in the day, and I am undecided about something to include in the release, so I'm going to sleep on it.
> 
> I just deployed a performance fix to the opportunity search. Wow, my head is spinning. I want to include it in the release, but it's a significant code change that hasn't been exercised much yet.
> 
> If you are inclined, you could test out the 2.5-rc branch and try to break the opportunity search.

I would be happy to test, but my schedule is pretty packed for the next week. I do have a client who makes use of the opportunity search pretty heavily, so I'd be more happy to dig for issues with that before it gets released. I can test next week, though!

Thanks @ginkgomzd !

**Comment by karenannx** [comment]
Hi Michael, just checking in. Any updates on the release?

It is very urgent for me, and most critically the ability to produce the reports.

 

Thanks so much,

Karen-Ann

**Comment by ginkgomzd** [comment]
Hi @karenannx ,
I think we are ready for a release, but I'm expecting a lot of people will eagerly install the new release, so I'm wanting to have a high confidence in the testing of the release.
I've got a daunting week ahead of me after the holiday last week.

I don't think you should be waiting for the release. The only thing the release will give you is a quicker upgrade. You should download this zip and extract it and replace your existing extension files.
https://github.com/civicrm/org.civicrm.volunteer/archive/2.5-rc.zip

If you would like to pay us to do this, you can email support@ginkgostreet.com, and we will arrange for you to securely provide your account credentials to us.

Cheers

**Comment by karenannx** [comment]
Hi Michael:

I downloaded the release and copied it in place, and I still receive the same error.

I suspect it has to do with that I am in joomla.

Any idea how much it would cost to find a fix for us joomla users?

 

Best,

Karen-Ann

**Comment by kelizoliva** [comment]
I have just fully tested the 2.5-rc branch on Drupal 6.67 with CiviCRM 5.13.4 (I know, I know, core is a little behind) for my client, and have just sent it to them for further testing. 

I will follow up if they happen to break the opportunity search, otherwise I think this can be merged into core. I will follow here once I have their feedback, which should be in the next few days.

**Comment by ginkgomzd** [comment]
Thanks @kelizoliva !
We've found a regression in the 2.5-rc and are working on a fix.
I'm leaning towards releasing the 2.5-rc branch over the 2.4-rc (numbering is really irrelevant), since I think we are giving it significant testing.

We are getting close!

**Comment by kelizoliva** [comment]
Just an update that our client has fully tested 2.5-rc and reports back that it addresses all the issues that they were having, and that they do not see any new issues thus far, so I am applying it to their live site. They have a pretty volunteer-heavy event coming up in two weeks, and will let me know if they run into any issues.

**Comment by ginkgomzd** [comment]
That's great, thanks @kelizoliva 

I know this keeps dragging out, but there are at least two more bug fixes in our private-fork that I want to get into the release. Since those fixes are getting a good testing in our branch, I have confidence to include them in the release. It is on my agenda to do the release this week, but I'm not good at under-promising...

**Comment by karenannx** [comment]
Hell all, any news on a new release? I am at a standstill, for 2 months.

**Comment by ginkgomzd** [comment]
Hi @karenannx ,
I don't think the release is what you are waiting for since you have already deployed the code in the release. You need a developer to look at the problem you are having on your instance. The "release" is technically just an announcement and a listing in the directory. 

Is it still this report issue that you are looking for a fix for?
https://civicrm.stackexchange.com/questions/30389/civivolunteer-error-after-upgrade-to-civi-5-11-0-on-joomla
If so, then either your deployment was not successful, or you have a cache problem. I'm starting to suspect that there are problems in a relatively new caching mechanism. You can look for some files in templates_c that contain "CachedCiviContainer..." in the filename. you can safely delete any files in templates_c.

If you send an email to support@ginkgostreet.com we would be able to provide you paid consulting for the issue on your instance. Alternatively, you can find any developer to take a detailed look at your problem and file a bug report, or even better, a pull-request for the fix.

**Comment by karenannx** [comment]
It was a good thought, I deleted all files in templates_c that had prefix “CachedCiviContainer” and still have the same problem / error.

I had sent an email to support@ginkgostreet.com <mailto:support@ginkgostreet.com> , I am adding that email to the “To”, I am hoping it is not a big issue, but it seems to be more of affix that is needed.

 

Best,

Karen-Ann

**Comment by MikeyMJCO** [comment]
Any update on this/a release for CiviVolunteer?

**Comment by ginkgomzd** [comment]
The status is the same. We haven't had the resources to do adequate testing. As documented in this thread, any confirmation of testing done is appreciated. Testing is the only thing holding up the release. I believe we have a good release candidate in the 2.5-rc branch.

Is there something else you are asking about @MikeyMJCO ?

**Comment by MikeyMJCO** [comment]
No, that's great! I'll find some time next week to do some testing on WP and D8 with the latest CiviCRM stable and the master branch.

**Comment by sunnycs121** [comment]
@ginkgomzd I am having same problem of volunteer opportunities, new volunteer project pages using front-end theme, was that fixed?

CMS: Drupal 7.67
CiviCRM: 5.13.6
CiviVolunteer: 4.7.31-2.3.1

**Comment by eileenmcnaughton** [comment]
@ginkgomzd I'm a bit worried that  we are putting  a  copy of civivolunteer in our demo builds  -  I would think it's OK except that we are putting out  a version that you are not comfortable tagging as a release yet - which makes  me wonder? My  personal feeling is that it probably IS tested by lots of sites using it but I think it  you really still aren't comfortable tagging it  as a release we should maybe remove if from the civibuild development & demo sites?

**Comment by ginkgomzd** [comment]
Hi @eileenmcnaughton ,
Maybe it is time to just tag the master branch and release it.
I was wanting to release either the 2.4-rc or 2.5-rc. I just tested both and found issues.

I'm guessing that the demo is using the master branch? I agree that that has been significantly been tested informally and would be okay to release.

If you say so, I'll tag the head of master as a release, and work on resolving the issues in the rc.

**Comment by eileenmcnaughton** [comment]
@ginkgomzd yeah I think you should. I'm pretty sure the code is BETTER than the current release from all the comments - you might just need to be a bit prepared to respond if it shakes out any new PRs

**Comment by eileenmcnaughton** [comment]
@ginkgomzd you can also mitigate a bit by incrementing the supported Civi version to 5.18 - then you will slow the adoption flow & reduce the risk that something doesn't work on all versions

**Comment by joemcl** [comment]
re testing, see #528

**Comment by eileenmcnaughton** [comment]
Note that on  current master the released version now gives notices - I just added #532

**Comment by ginkgomzd** [comment]
Hello issue-followers.
Since this has been bumped again, I want to call again for assistance testing.
This is the biggest obstacle for completing a release. 
I could alternatively, be nudged into just stamping the current master branch, if I get a plurality of feedback that it is reasonably safe and an improvement on having no recent release.

If we want to aim a little higher, there is a branch [2.5-rc] and a PR [#548] that need testing. 

I think if I get bug reports with steps to reproduce, I could manage the fixes.

I'm still in a rough space, but someday this ordeal will be over (yes, this is the scale of years). I am willing to hand over the maintainer title to someone who demonstrates the ability with my oversight first. I won't stand in the way of a new maintainer taking over.

Salud

**Comment by MikeyMJCO** [comment]
I've booked in time to run RC branch testing and we've got some automated testing setup on a fork that we're refining. I haven't forgotten this but it does get nudged to the back of the queue routinely :-)

**Comment by nickperkins** [comment]
Old issue but still open. Is there a reason there hasn't been a release in a couple of years?

**Comment by ginkgomzd** [comment]
@nickperkins Hi.
This thread should have been closed.

There was a release in May, so not a couple years. There is not active development on this extension. 
My current thinking is that the extension should be broken apart into smaller extensions and parts re-written... for various reasons. Please get in touch if you are interested in further development.
